src/restore.ts

```diff
@@ -4,69 +4,129 @@ import * as exec from "@actions/exec";
+    }
   } else {
     core.info("No cache found.");
+    if (SELF_CI) {
+      core.setOutput("test-cache-hit", false)
+    }
   }
 }
 
-async function configure() {
+async function configure(ccacheVariant : string) : Promise<void> {
   const ghWorkSpace = process.env.GITHUB_WORKSPACE;
   const maxSize = core.getInput('max-size');
   
-  core.info("Configure ccache");
-  await exec.exec("ccache --set-config=cache_dir=" + ghWorkSpace + "/.ccache");
-  await exec.exec("ccache --set-config=max_size=" + maxSize);
-  await exec.exec("ccache --set-config=compression=true");
+  core.info(`Configure ${ccacheVariant}`);
+  if (ccacheVariant === "ccache") {
+    await exec.exec(`ccache --set-config=cache_dir=${ghWorkSpace}/.ccache}`);
+    await exec.exec(`ccache --set-config=max_size=${maxSize}`);
+    await exec.exec(`ccache --set-config=compression=true`);
+    core.info("Cccache config:");
+    await exec.exec("ccache -p");
+  } else {
+    const options = `SCCACHE_IDLE_TIMEOUT=999999 SCCACHE_DIR="${ghWorkSpace}"/.sccache SCCACHE_CACHE_SIZE="${maxSize}"`;
+    await exec.exec(`env ${options} sccache --start-server`);
+  }
+
+}
+
+async function installCcacheMac() : Promise<void> {
+  await exec.exec("brew install ccache");
+}
+
+async function installCcacheLinux() : Promise<void> {
+  await exec.exec("sudo apt-get install -y ccache");
+}
+
+async function installCcacheWindows() : Promise<void> {
+  throw Error("Ccache is not available for Windows, use Sccache with 'sccache: true'")
+}
+
+async function installSccacheMac() : Promise<void> {
+  await exec.exec("brew install sccache");
+}
+
+async function installSccacheLinux() : Promise<void> {
+  await installSccacheFromGitHub(
+    "v0.2.15",
+    "x86_64-unknown-linux-musl",
+    "e5d03a9aa3b9fac7e490391bbe22d4f42c840d31ef9eaf127a03101930cbb7ca",
+    "/usr/local/bin/",
+    "sccache"
+  );
+}
+
+async function installSccacheWindows() : Promise<void> {
+  await installSccacheFromGitHub(
+    "v0.2.15",
+    "x86_64-pc-windows-msvc",
+    "e5d03a9aa3b9fac7e490391bbe22d4f42c840d31ef9eaf127a03101930cbb7ca",
+    // TODO find a better place
+    "C:\\Users\\runneradmin\\.cargo\\bin",
+    "sccache.exe"
+  );
+}
 
-  core.info("Ccache config:")
-  await exec.exec("ccache -p");
+async function installSccacheFromGitHub(version : string, artifactName : string, sha256 : string, binPath : string, binName : string) : Promise<void> {
+  const archiveName = `sccache-${version}-${artifactName}`;
+  const url = `https://github.com/mozilla/sccache/releases/download/${version}/${archiveName}.tar.gz`;
+  await exec.exec("sh", ["-c", `curl -L '${url}' | tar xzf - -O '${archiveName}/${binName}' > '${binPath}/${binName}'`]);
+  await exec.exec(`echo "${sha256}  ${binPath}/${binName}" | sha256sum -c`);
+  await exec.exec(`chmod +x ${binPath}/${binName}`);
 }
 
 async function run() : Promise<void> {
   try {
-    let ccachePath = await io.which("ccache");
+    const useSccache = core.getBooleanInput("sccache");
```

Comment:
`sccache: true`

would it be possible to change this to:

`variant: sccache`

in the config? You use it `ccacheVariant` below, so why not?

Reply:
Sounds good!